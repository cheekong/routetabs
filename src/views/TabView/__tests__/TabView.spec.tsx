import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import { TabView } from "..";

const renderComponent = (pathname: string) => {
  return render(
    <MemoryRouter initialEntries={[{ pathname }]}>
      <TabView />
    </MemoryRouter>
  );
};

describe("TabList", () => {
  it("Test that the correct tabpanel for regular purple potato renders via url changes", async () => {
    const { container } = renderComponent("/potato/purple");

    userEvent.click(screen.getByTestId("regular"));
    expect(screen.getByText("TBA. Purples rolling in")).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it("Test that the correct tabpanels for noodle renders via url changes", async () => {
    const { container } = renderComponent("/noodle");

    userEvent.click(screen.getByTestId("noodle"));
    expect(screen.getByText("more noodles to come")).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
