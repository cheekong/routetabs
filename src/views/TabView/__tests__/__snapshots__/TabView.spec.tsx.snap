// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList Test that the correct tabpanel for regular purple potato renders via url changes 1`] = `
<div>
  <main
    id="tab-view"
  >
    <div
      id="tab-content"
    >
      <h1>
        Router Tabs
      </h1>
      <div
        class="tabs"
        data-testid="carbTabs"
        id="carbTabs"
      >
        <div
          aria-label="carbTabList"
          data-testid="carbTabList"
          role="tablist"
        >
          <button
            aria-controls="potatoTabPanel"
            aria-selected="true"
            data-testid="potato"
            id="potato"
            role="tab"
            tabindex="0"
            type="button"
          >
            Potato
          </button>
          <button
            aria-controls="noodleTabPanel"
            aria-selected="false"
            data-testid="noodle"
            id="noodle"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Noodle
          </button>
        </div>
        <div
          aria-labelledby="potato"
          data-testid="potatoTabPanel"
          id="potatoTabPanel"
          role="tabpanel"
          tabindex="0"
        >
          <div
            aria-label="potatoTabList"
            aria-labelledby="potatoTabPanel"
            data-testid="potatoTabList"
            role="tablist"
          >
            <button
              aria-controls="sweetPanel"
              aria-selected="false"
              data-testid="sweet"
              id="sweet"
              role="tab"
              tabindex="0"
              type="button"
            >
              sweet
            </button>
            <button
              aria-controls="purplePanel"
              aria-selected="true"
              data-testid="purple"
              id="purple"
              role="tab"
              tabindex="0"
              type="button"
            >
              purple
            </button>
          </div>
          <div
            aria-labelledby="purple"
            data-testid="purplePanel"
            id="purplePanel"
            role="tabpanel"
            tabindex="0"
          >
            <div
              aria-label="potatoTabList"
              aria-labelledby="potatoTabPanel"
              data-testid="potatoTabList"
              role="tablist"
            >
              <button
                aria-controls="japanesePanel"
                aria-selected="false"
                data-testid="japanese"
                id="japanese"
                role="tab"
                tabindex="0"
                type="button"
              >
                japanese purple potato
              </button>
              <button
                aria-controls="regularPanel"
                aria-selected="true"
                data-testid="regular"
                id="regular"
                role="tab"
                tabindex="0"
                type="button"
              >
                regular purple potato
              </button>
            </div>
            <div
              aria-labelledby="regular"
              data-testid="regularPanel"
              id="regularPanel"
              role="tabpanel"
              tabindex="0"
            >
              TBA. Purples rolling in
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`TabList Test that the correct tabpanels for noodle renders via url changes 1`] = `
<div>
  <main
    id="tab-view"
  >
    <div
      id="tab-content"
    >
      <h1>
        Router Tabs
      </h1>
      <div
        class="tabs"
        data-testid="carbTabs"
        id="carbTabs"
      >
        <div
          aria-label="carbTabList"
          data-testid="carbTabList"
          role="tablist"
        >
          <button
            aria-controls="potatoTabPanel"
            aria-selected="false"
            data-testid="potato"
            id="potato"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Potato
          </button>
          <button
            aria-controls="noodleTabPanel"
            aria-selected="true"
            data-testid="noodle"
            id="noodle"
            role="tab"
            tabindex="0"
            type="button"
          >
            Noodle
          </button>
        </div>
        <div
          aria-labelledby="noodle"
          data-testid="noodleTabPanel"
          id="noodleTabPanel"
          role="tabpanel"
          tabindex="0"
        >
          more noodles to come
        </div>
      </div>
    </div>
  </main>
</div>
`;
